src/nunit.analyzers/Helpers/AssertExpressionHelper.cs

```diff
@@ -52,32 +53,58 @@ internal static class AssertExpressionHelper
 
             foreach (var constraintPart in constraintParts)
             {
-                var invocations = constraintPart.SplitCallChain()
-                    .OfType<InvocationExpressionSyntax>()
-                    .Where(i => i.ArgumentList.Arguments.Count == 1);
+                var expected = GetExpectedArgumentFromConstraintPart(constraintPart, semanticModel, cancellationToken);
 
-                foreach (var invocation in invocations)
+                if (expected != null)
                 {
-                    var symbol = semanticModel.GetSymbolInfo(invocation, cancellationToken).Symbol;
-
-                    if (symbol is IMethodSymbol methodSymbol
-                        && methodSymbol.Parameters.Length == 1
-                        && methodSymbol.Parameters[0].Name == NunitFrameworkConstants.NameOfExpectedParameter)
-                    {
-                        var argument = invocation.ArgumentList.Arguments[0];
-                        expectedArguments.Add((argument.Expression, methodSymbol));
-                    }
+                    expectedArguments.Add(expected.Value);
                 }
             }
 
             return expectedArguments;
         }
 
+        /// <summary>
+        /// Split constraints into parts by binary ('&' or '|')  or constraint expression operators
+        /// </summary>
+        public static IEnumerable<ExpressionSyntax> SplitConstraintByOperators(ExpressionSyntax constraintExpression)
+        {
+            return SplitConstraintByBinaryOperators(constraintExpression)
+                .SelectMany(SplitConstraintByConstraintExpressionOperators);
+        }
+
+        public static IEnumerable<ExpressionSyntax> GetConstraintExpressionPrefixes(
+            ExpressionSyntax constraintExpression, SemanticModel semanticModel)
+        {
+            // e.g. Has.Property("Prop").Not.EqualTo("1")
+            // -->
+            // Has.Property("Prop"),
+            // Has.Property("Prop").Not
+
+            // Take expressions until found expression returning constraint
+            return GetExpressionsFromCurrentPart(constraintExpression)
+                .Where(e => e is MemberAccessExpressionSyntax || e is InvocationExpressionSyntax)
+                .TakeWhile(e => !ReturnsConstraint(e, semanticModel));
+        }
+
+        public static IEnumerable<ExpressionSyntax> GetConstraintExpressionSuffixes(
+            ExpressionSyntax constraintExpression, SemanticModel semanticModel)
+        {
+            // e.g. Has.Property("Prop").Not.EqualTo("1").IgnoreCase
+            // -->
+            // Has.Property("Prop").Not.EqualTo("1").IgnoreCase
```

Comment:
Probably not the best example, but yes.
Method will return either `MemberAccessExpression` (in case of property), or `InvocationExpression` (in case of method).

In this case `Name` property of expression would be `IgnoreCase`, but the whole expressions would be, well, the whole expression.